Library/Homebrew/language/node.rb

```diff
@@ -1,10 +1,44 @@
+require "json"
+
 module Language
   module Node
     def self.npm_cache_config
       "cache=#{HOMEBREW_CACHE}/npm_cache"
     end
 
-    def self.pack_for_installation
+    def self.node_system(cmd, *args)
```

Comment:
And speaking about `formula`: Is it possible to access the `formula`'s download `url` inside `language/node`? With it we could match it against a npm registry url regex and figure the correct way to handle life cycle scripts out on our own without requiring the formula author to set the `prepare_require` parameter.